jobbergate-docs/docs/source/developer_guide/integration_testing.md

```diff
@@ -0,0 +1,533 @@
+jobbergate job-submissions create --name=$NAME --job-script-id=1 --cluster-name=local-slurm --execution-directory=/nfs
+```
+
+Verify that the output shows that the Job Submission has been created for the target
+Job Script
+
+
+```
+                  Created Job Submission
+┏━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+┃ Key                 ┃ Value                            ┃
+┡━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
+│ id                  │ 1                                │
+│ job_script_id       │ 1                                │
+│ cluster_name        │ local-slurm                      │
+│ slurm_job_id        │ None                             │
+│ execution_directory │ /nfs                             │
+│ name                │ test--root--2023-10-13           │
+│ description         │                                  │
+│ owner_email         │ local-user@jobbergate.local-mail │
+│ status              │ CREATED                          │
+│ report_message      │ None                             │
+└─────────────────────┴──────────────────────────────────┘
+```
+
+
+### Verify that the agent submitted the job
+
+To verify that the agent submitted the job correctly, review the log output from the
+agent.
+
+The agent performs the following process to complete a Job Submission with Slurm
+
+ - Fetch a pending job from the API
+ - Submit the job to slurm
+ - Mark the job as submitted and add the slurm job id to it
+
+You can access the log data by running the following command in a terminal that has
+changed directory to the `jobbergate-composed` folder:
+
+```shell
+docker-compose logs jobbergate-agent
+```
+
+It may be useful to pipe the output to a text viewer like `less`.
+
+If the agent has successfully submitted the job, you should see some log lines that
+look like this (ellipses indicate omitted content):
+
+```
+...Retrieved 1 pending job submission...
+...Submitting pending job submission 1
+...
+...Submitting pending job submission 1 to slurm...
+...
+...Received slurm job id 1 for job submission 1
+...Marking job job_submission_id=22 as SUBMITTED (slurm_job_id=1)
+```
+
+If you find those log lines, then the Agent has successfully submitted the job to slurm.
+
+
+### Verify that the job was completed
+
+To verify that the job completed successfully, review the log output from the
+agent.
+
+The agent performs the following process to complete Job Submissions
+
+ - Fetch the submitted job from the API
+ - Check the status of the job in slurm using the the `slurm_job_id`
+ - If the job is completed, mark the Job Submission as COMPLETED
+
+You should look for log lines that look like this (ellipses indicate omitted content):
+
+```
+...Retrieved 1 active job submissions...
+...Fetching status of job_submission 1 from slurm
+...Fetching slurm job status for slurm job 1
+...
+...Status for slurm job 1 is job_id=1 job_state='COMPLETED'...
+...Updating job_submission with status=COMPLETED
+```
+
```

Comment:
Getting job-submissions may be more friendly than watching the agent's logs. The downside is that we can not ensure the ID in advance.

```bash
watch jobbergate get-job-submission --id <job_submissions_id>
```

Reply:
In this case, I like introducing the idea of watching logs because it informs the dev where the logs can be seen and it will be helpful if something is going wrong to see it in the logs themselves.